File: m3u8_downloader/Models/VideoFile.cs
using System;
using Prism.Mvvm;

namespace m3u8_downloader.Models
{
    public class VideoFile : BindableBase
    {
        /// <summary>
        /// 封面
        /// </summary>
        public string CoverImage { set; get; }

        /// <summary>
        /// 视频名
        /// </summary>
        private string _videoName;

        public string VideoName
        {
            get => _videoName;
            set => SetProperty(ref _videoName, value);
        }

        /// <summary>
        /// 文件路径
        /// </summary>
        private string _filePath;

        public string FilePath
        {
            get => _filePath;
            set => SetProperty(ref _filePath, value);
        }

        /// <summary>
        /// 分辨率
        /// </summary>
        public string Resolution { set; get; }

        /// <summary>
        /// 视频大小
        /// </summary>
        public string VideoSize { set; get; }

        /// <summary>
        /// 时长
        /// </summary>
        public string Duration { set; get; }

        /// <summary>
        /// 文件修改时间，用于缓存有效性检查
        /// </summary>
        public DateTime LastModified { get; set; }
    }
}

File: lib/models/video_file.dart
class VideoFile {
  /// 封面
  String coverImage;

  /// 视频名
  String videoName;

  /// 文件路径
  String filePath;

  /// 分辨率
  String resolution;

  /// 视频大小
  String videoSize;

  /// 时长
  String duration;

  /// 文件修改时间，用于缓存有效性检查
  DateTime lastModified;

  // 构造函数
  VideoFile({
    required this.coverImage,
    required this.videoName,
    required this.filePath,
    required this.resolution,
    required this.videoSize,
    required this.duration,
    required this.lastModified,
  });
}
